src/pages/projectSettings/tabs/NotificationsTab/SubscriptionField.tsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import styled from "@emotion/styled";
+import ExpandableCard from "@leafygreen-ui/expandable-card";
+import { Body } from "@leafygreen-ui/typography";
+
+import { Field } from "@rjsf/core";
+import { size } from "constants/tokens";
+import { triggers } from "constants/triggers";
+import { Subscriber } from "gql/generated/types";
+import { ResourceType, Trigger } from "types/triggers";
+import { SubscriptionSelect } from "./NotificationsRow";
+
+const getSentanceCase = (string: string) =>
+  string[0].toUpperCase() + string.substring(1).toLowerCase();
+
+export const getSubscriberType = (subscriber: Subscriber) => {
+  if (subscriber === undefined) {
+    return "";
+  }
+  let key: string = "";
+  Object.keys(subscriber).forEach((k) => {
+    if (subscriber[k] !== null) {
+      key = k;
+      return k;
+    }
+    return null;
+  });
+  return key;
+};
+
+export const getTrigger = (trigger, resourceType: string) => {
+  let triggerData: Trigger = {
+    trigger: "",
+    resourceType: ResourceType.VERSION,
+    regexSelectors: [],
+    label: "",
+  };
+  if (trigger === undefined) {
+    return triggerData;
+  }
+  triggers.forEach((t) => {
+    console.log("trigger: ", JSON.stringify(trigger));
+    if (t.trigger === trigger && t.resourceType === resourceType) {
+      triggerData = t;
+      return t;
+    }
+  });
+  return triggerData;
+};
```

Comment:
You may not have to explicitly check for the `undefined` case since I think it will fall through. Also, seems like there's an extra `console.log` statement here
```suggestion
  triggers.forEach((t) => {
    if (t.trigger === trigger && t.resourceType === resourceType) {
      return t;
    }
  });
  return triggerData;
};
```